haskell/private/ghci_repl_wrapper.sh

```diff
@@ -52,13 +52,17 @@ cd "$BUILD_WORKSPACE_DIRECTORY"
 # location of exec root reliably and then prefix locations of various
 # components, such as shared libraries with that exec root.
 
-RULES_HASKELL_EXEC_ROOT=$(dirname $(readlink ${BUILD_WORKSPACE_DIRECTORY}/bazel-out))
-TOOL_LOCATION="$RULES_HASKELL_EXEC_ROOT/{TOOL}"
+RULES_HASKELL_EXEC_ROOT=$(dirname "$(readlink "${BUILD_WORKSPACE_DIRECTORY}"/bazel-out)")
+TOOL_LOCATION="$RULES_HASKELL_EXEC_ROOT/%{TOOL}"
 # Setting -pgm* flags explicitly has the unfortunate side effect
 # of resetting any program flags in the GHC settings file. So we
 # restore them here. See
 # https://ghc.haskell.org/trac/ghc/ticket/7929.
-PGM_ARGS="-pgma {CC} -pgmc {CC} -pgml {CC} -pgmP {CC} -optc-fno-stack-protector -optP-E -optP-undef -optP-traditional"
+PGM_ARGS="-pgma %{CC} -pgmc %{CC} -pgml %{CC} -pgmP %{CC} -optc-fno-stack-protector -optP-E -optP-undef -optP-traditional"
 
-{ENV}
-"$TOOL_LOCATION" $PGM_ARGS {ARGS} "$@"
+# shellcheck disable=SC1083
+%{ENV}
+
+# Disabling 2086, because we indeed want word splitting on PGM_ARGS
```

Comment:
PGM_ARGS should be a bash array then.

Reply:
Done in 46f554b41439c87de16cf6f0f17929dfe0251a5e. All these changes to bash arrays are nice, the code is clearer and shellcheck warnings went away. :+1: